ports/rp2/mpconfigport.h

```diff
@@ -107,10 +109,22 @@
 #define MICROPY_FATFS_ENABLE_LFN                (1)
 #define MICROPY_FATFS_LFN_CODE_PAGE             437 /* 1=SFN/ANSI 437=LFN/U.S.(OEM) */
 #define MICROPY_FATFS_RPATH                     (2)
+#if MICROPY_HW_USB_MSC
+#define MICROPY_FATFS_USE_LABEL                 (1)
+#define MICROPY_FATFS_MULTI_PARTITION           (1)
+// Set FatFS block size to flash sector size to avoid caching
+// the flash sector in memory to support smaller block sizes.
+#define MICROPY_FATFS_MAX_SS                    (FLASH_SECTOR_SIZE)
```

Comment:
Defining this as 4096 will mean that all VfsFat objects now allocate 4096 bytes of buffer space (for the internals of the FAT driver).  So it doesn't really save RAM, even though a cache is not needed.

That said, it's probably the right approach to use a 4096 block size, to avoid difficulties configuring the block size when the filesystem type changes (eg going from littlefs to FAT, stm32 has a lot of complex code to deal with changing the block size between 4096 and 512).

Reply:
Yes, but it's allocated only for the `FATFS` object and not per file (as long as `FF_FS_TINY` is 1). Note the same buffer size will be reserved either way if caching is used (plus 512 bytes for `MICROPY_FATFS_MAX_SS`). FWIW I tried the caching approach, it's doable, but I don't see the need to make this code more complex than it is right now.